services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -1038,59 +1041,63 @@ export class WorkflowService {
     const snapshot = service.getSnapshot();
     const currentState = snapshot.value;
     const context = snapshot.machine.context;
+    // TODO: Refactor to use snapshot.done instead
     const isFinal = snapshot.machine.states[currentState].type === 'final';
+    const entityType = context.entity.type === 'individual' ? 'endUser' : ('business' as const);
+    const entityId = runtimeData[`${entityType}Id`];
 
     this.logger.log('Workflow state transition', {
       id: id,
       from: runtimeData.state,
       to: currentState,
     });
 
-    // TODO: Update to work with changes related to revision
-    if (type === 'resubmit' && document) {
-      switch (resubmissionReason) {
-        case ResubmissionReason.BLURRY_IMAGE:
-          await this.workflowRuntimeDataRepository.updateById(
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-explicit-any
-            (runtimeData as any).context?.parentMachine?.id,
-            {
-              data: {
-                state: 'document_photo',
-                status: 'active',
-                context: {
-                  ...context,
-                  [document]: {
-                    ...context?.[document],
-                    resubmissionReason,
-                  },
-                },
-              },
-            },
-          );
-          break;
-        default:
-          throw new BadRequestException(
-            `Invalid resubmission reason ${resubmissionReason as string}`,
-          );
-      }
-    }
-
-    await this.updateWorkflowRuntimeData(runtimeData.id, {
+    const updatedRuntimeData = await this.updateWorkflowRuntimeData(runtimeData.id, {
       context,
       state: currentState,
       status: isFinal ? 'completed' : runtimeData.status,
     });
 
+    this.workflowEventEmitter.emit('workflow.state.changed', {
+      entityId,
+      state: updatedRuntimeData.state,
+      correlationId: updatedRuntimeData.correlationId,
+      runtimeData: updatedRuntimeData,
+    });
+
     if (!isFinal || (currentState !== 'approved' && currentState !== 'rejected')) {
       return;
     }
 
-    runtimeData.endUserId &&
-      (await this.endUserRepository.updateById(runtimeData.endUserId, {
+    const approvalState = ApprovalState[currentState.toUpperCase() as keyof typeof ApprovalState];
+
+    if (!entityType) {
+      throw new BadRequestException(`entity.type is required`);
+    }
+
+    if (!entityId) {
+      throw new BadRequestException(`entity.${entityType}Id is required`);
+    }
+
+    if (entityType === 'endUser') {
```

Comment:
extract to a common util, something like "update entity" which will check the type (when in the future it will be refactored so it will not need to know the type it will just update BallerineEntityId)

Reply:
Added a temporary entity repository. The tech debt is now finding a way of not breaking TypeScript when not checking `entityType`'s value.